reporters/kamon-opentelemetry/src/main/scala/kamon/otel/OpenTelemetryTraceReporter.scala

```diff
@@ -88,21 +86,26 @@ class OpenTelemetryTraceReporter(traceServiceFactory:Config=>TraceService)(impli
   }
 
   /**
-   * Builds the resource information added as resource labels to the exported traces
-   * @param includeEnvTags
-   * @return
-   */
-  private def buildResource(includeEnvTags:Boolean):Resource = {
+    * Builds the resource information added as resource labels to the exported traces
+    *
+    * @param includeEnvTags
+    * @return
+    */
+  private def buildResource(includeEnvTags: Boolean): Resource = {
     val env = Kamon.environment
-    val builder = Resource.newBuilder()
-      .addAttributes(stringKeyValue("service.name", env.service))
-      .addAttributes(stringKeyValue("telemetry.sdk.name", "kamon"))
-      .addAttributes(stringKeyValue("telemetry.sdk.language", "scala"))
-      .addAttributes(stringKeyValue("telemetry.sdk.version", kamonVersion))
+    val builder = Resource.builder()
+      .put("service.name", env.service)
+      .put("telemetry.sdk.name", "kamon")
+      .put("telemetry.sdk.language", "scala")
+      .put("telemetry.sdk.version", kamonVersion)
 
     //add all kamon.environment.tags as KeyValues to the Resource object
-    if(includeEnvTags) {
-      env.tags.iterator().map(toProtoKeyValue).foreach(builder.addAttributes)
+    if (includeEnvTags) {
+      env.tags.iterator().foreach {
+        case t: Tag.String => builder.put(t.key.replace('-', '.'), t.value)
+        case t: Tag.Boolean => builder.put(t.key.replace('-', '.'), t.value)
+        case t: Tag.Long => builder.put(t.key.replace('-', '.'), t.value)
+      }
```

Comment:
I think this is not necessary since [v2.4.6](https://github.com/kamon-io/Kamon/releases/tag/v2.4.6). Can you please confirm and remove if that's the case?